src/Rules/Functions/RandomIntParametersRule.php

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\IntegerRangeType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\VerbosityLevel;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\FuncCall>
+ */
+class RandomIntParametersRule implements \PHPStan\Rules\Rule
+{
+
+	/** @var ReflectionProvider */
+	private $reflectionProvider;
+
+	public function __construct(ReflectionProvider $reflectionProvider)
+	{
+		$this->reflectionProvider = $reflectionProvider;
+	}
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof \PhpParser\Node\Name)) {
+			return [];
+		}
+
+		if ($this->reflectionProvider->resolveFunctionName($node->name, $scope) !== 'random_int') {
+			return [];
+		}
+
+		$minType = $scope->getType($node->args[0]->value)->toInteger();
+		$maxType = $scope->getType($node->args[1]->value)->toInteger();
+
+		if ($minType->equals(new IntegerType()) || $maxType->equals(new IntegerType())) {
+			return [];
+		}
+
+		if ($minType instanceof ConstantIntegerType || $minType instanceof IntegerRangeType) {
+			if ($minType instanceof ConstantIntegerType) {
+				$maxPermittedType = IntegerRangeType::fromInterval($minType->getValue(), PHP_INT_MAX);
+			} else {
+				$maxPermittedType = IntegerRangeType::fromInterval($minType->getMax(), PHP_INT_MAX);
+			}
+
+			if (!$maxPermittedType->isSuperTypeOf($maxType)->yes()) {
+				$message = 'Cannot call random_int() when $min parameter (%s) can be greater than $max parameter (%s).';
+
+				if ($maxType->isSuperTypeOf($minType)->no()) {
+					$message = 'Cannot call random_int() when $min parameter (%s) is greater than $max parameter (%s).';
```

Comment:
These messages should be more consistent with how this is usually reported, for example:

```
Parameter #1 $depth of function xdebug_call_class expects int, string given.
```

So something like this?

```
Parameter #1 $min (int<min, 5>) of function random_int expects lower number than parameter #2 $max (-1).
```